---
title: "1_mapping_to_genome"
author: "fangting"
date: "2019/9/30"
output: html_document
editor_options: 
  chunk_output_type: console
---

# mapping to genome
    
## hisat2 mapping reads to refgenome

```{r,eval=FALSE}
## download gff3 of maize and convert gff3 file into gtf file
gffread Zea_mays.B73_RefGen_v4.41.gff3 -T -o Zea_mays.B73_RefGen_v4.41.gtf

## extract_exons.py of hisat2 extracts exons
extract_exons.py Zea_mays.B73_RefGen_v4.41.gtf > Zea_mays.B73_RefGen_v4.41.exons

## extract_splice_sites.py of hisat2 extracts splice sites
extract_splice_sites.py Zea_mays.B73_RefGen_v4.41.gtf > Zea_mays.B73_RefGen_v4.41.splicesites

## hisat2-build for index
hisat2-build -p 8 Zea_mays.AGPv4.dna.toplevel.fa --exon Zea_mays.B73_RefGen_v4.41.exons --ss Zea_mays.B73_RefGen_v4.41.splicesites Zea_mays.B73_RefGen_v4.41_tran

#$ -S /bin/bash
#$ -j y
#$ -V
#$ -cwd
#$ -l h_vmem=100G
hisat2 -x Zea_mays.B73_RefGen_v4.41_tran --rna-strandness RF -1 path/to/cleandata/D2_1_input_CleanData_R1.fastq.gz -2 path/to/cleandata/cleandata/D2_1_input_CleanData_R2.fastq.gz -S path/to/sam/D2_1_input.sam --summary-file D2_1_input.txt
```

## samtools converts SAM into BAM and filtered

```{r,eval=FALSE}
#$ -S /bin/bash
#$ -j y
#$ -V
#$ -cwd
#$ -l h_vmem=100G
samtools view -bS path/to/sam/D2_1_input.sam | samtools view -h -q 50 | grep 'NH:i:1' -> path/to/filtered_bam/D2_1_input.filtered.bam
```

