---
title: "2_htseq_for_reads_count"
author: "fangting"
date: "2019/9/30"
output: html_document
editor_options: 
  chunk_output_type: console
---

# htseq-count for reads count of input samples

## samtools sorts bam by position,
## the parameter of -T used to specify temporary file, note that the temporary files for each sample cannot be in the same directory

```{r,eval=FALSE}
#$ -S /bin/bash
#$ -j y
#$ -V
#$ -cwd
#$ -l h_vmem=100G
samtools sort -T tmp.bam path/to/filtered_bam/D2_1_input.filtered.bam -o path/to/pos_sorted_bam/D2_1_input_pos_sorted.bam
```

## htseq-count calculates reads of bam file
## the -s parameter is setted as reverse for the chain-specific coding genes

```{r,eval=FALSE}
#$ -S /bin/bash
#$ -j y
#$ -V
#$ -cwd
#$ -l h_vmem=100G
htseq-count -f bam -r pos -s reverse -a 10 -m union --nonunique none path/to/pos_sorted_bam/D2_1_input_pos_sorted.bam path/to/ref/Zea_mays.B73_RefGen_v4.41.gtf > path/to/reads/D2_1_input.htseq.count.pos.sorted.txt
```

