---
title: "4_exomePeak_call_peaks"
author: "fangting"
date: "2019/9/30"
output: html_document
editor_options: 
  chunk_output_type: console
---

# exomePeak calls peaks

```{r,eval=FALSE}
library(exomePeak)

## sample D2 peak calling
setwd(path/to/D2_sample_bam)

IP_BAM=c("D2_1_IP_pos_sorted.bam","D2_2_IP_pos_sorted.bam")
INPUT_BAM=c("D2_1_input_pos_sorted.bam","D2_2_input_pos_sorted.bam")
result=exomepeak(GENE_ANNO_GTF = "path/to/ref/Zea_mays.B73_RefGen_v4.41.gtf",
                 IP_BAM=IP_BAM,
                 INPUT_BAM=INPUT_BAM,
                 EXPERIMENT_NAME="D2_peak")

## sample D4 peak calling
setwd(path/to/D4_sample_bam)

IP_BAM=c("D4_1_IP_pos_sorted.bam","D4_2_IP_pos_sorted.bam")
INPUT_BAM=c("D4_1_input_pos_sorted.bam","D4_2_input_pos_sorted.bam")
result=exomepeak(GENE_ANNO_GTF = "path/to/ref/Zea_mays.B73_RefGen_v4.41.gtf",
                 IP_BAM=IP_BAM,
                 INPUT_BAM=INPUT_BAM,
                 EXPERIMENT_NAME="D4_peak")

## sample D8 peak calling
setwd(path/to/D8_sample_bam)

IP_BAM=c("D8_1_IP_pos_sorted.bam","D8_2_IP_pos_sorted.bam")
INPUT_BAM=c("D8_1_input_pos_sorted.bam","D8_2_input_pos_sorted.bam")
result=exomepeak(GENE_ANNO_GTF = "path/to/ref/Zea_mays.B73_RefGen_v4.41.gtf",
                 IP_BAM=IP_BAM,
                 INPUT_BAM=INPUT_BAM,
                 EXPERIMENT_NAME="D8_peak")

## sample N2 peak calling
setwd(path/to/N2_sample_bam)

IP_BAM=c("N2_1_IP_pos_sorted.bam","N2_2_IP_pos_sorted.bam")
INPUT_BAM=c("N2_1_input_pos_sorted.bam","N2_2_input_pos_sorted.bam")
result=exomepeak(GENE_ANNO_GTF = "path/to/ref/Zea_mays.B73_RefGen_v4.41.gtf",
                 IP_BAM=IP_BAM,
                 INPUT_BAM=INPUT_BAM,
                 EXPERIMENT_NAME="N2_peak")

## sample N4 peak calling
setwd(path/to/N4_sample_bam)

IP_BAM=c("N4_1_IP_pos_sorted.bam","N4_2_IP_pos_sorted.bam")
INPUT_BAM=c("N4_1_input_pos_sorted.bam","N4_2_input_pos_sorted.bam")
result=exomepeak(GENE_ANNO_GTF = "path/to/ref/Zea_mays.B73_RefGen_v4.41.gtf",
                 IP_BAM=IP_BAM,
                 INPUT_BAM=INPUT_BAM,
                 EXPERIMENT_NAME="N4_peak")

## sample N8 peak calling
setwd(path/to/N8_sample_bam)

IP_BAM=c("N8_1_IP_pos_sorted.bam","N8_2_IP_pos_sorted.bam")
INPUT_BAM=c("N8_1_input_pos_sorted.bam","N8_2_input_pos_sorted.bam")
result=exomepeak(GENE_ANNO_GTF = "path/to/ref/Zea_mays.B73_RefGen_v4.41.gtf",
                 IP_BAM=IP_BAM,
                 INPUT_BAM=INPUT_BAM,
                 EXPERIMENT_NAME="N8_peak")
```

Here, only the peaks of protein coding genes from nuclear DNA were retained. Further, the methylated genes with low transcript abundance were excluded. 

# peaks merge and sequences extract

```{r,eval=FALSE}
## merge peaks of protein coding genes in six samples
cat *peak.txt | sort -k1,1 -k2,2n > com.sort.peak.txt
bedtools merge -i com.sort.peak.txt -s -o distinct > allsample.com.peak.txt

## extract exon positions of each peak from canonical transcript 
for i in *.pos.txt
 do
  pro=`basename $i .pos.txt`
  bedtools intersect -a path/to/canonical_exons/maize.all.exon.txt -b $i -s > $pro.pos.exon.txt
 done
 
## merge all exons into a unique file
cat *.pos.exon.canonical.trans.txt > all.merge.peak.pos.exon.canonical.trans.txt

## gffread extracts the sequences of peak
gffread -w all.merge.peak.pos.exon.canonical.trans.fa -g path/to/ref/genome/Zea_mays.AGPv4.dna.toplevel.fa all.merge.peak.pos.exon.canonical.trans.txt
## filter out peak sequences less than 50bp
awk 'BEGIN {RS = ">" ; ORS = ""} length($2) >= 50 {print ">"$0}' all.merge.peak.pos.exon.canonical.trans.fa > all.merge.peak.pos.exon.canonical.trans.filter.fa
## transform DNA sequences into RNA sequences
less all.merge.peak.pos.exon.canonical.trans.filter.fa | sed 's/T/U/g' > all.merge.peak.pos.exon.canonical.trans.filter.RNA.fa
```